test_menu.js

```diff
@@ -0,0 +1,90 @@
+//Tests all menu blocks for expected behaviour.
+var url = "https://neontabs.neontribe.org/latest/hr/"
+var base_url = "https://neontabs.neontribe.org"
+var link_count = 0;
+var tests = link_count + 10;
+
+casper.start(casper.cli.get("target"), function(){
+  	//Count links
+	var menu_links = casper.evaluate(function() {
+			var links = document.querySelectorAll('ul.main-menu > li > a');
+			return Array.prototype.map.call(links, function(e) {
+    		return e.getAttribute("href");
+  		});
+  	});
+
+	var dropdown_links = casper.evaluate(function() {
+		var links = document.querySelectorAll('li.dropdown a');
+		return Array.prototype.map.call(links, function(e) {
+			return e.getAttribute("href");
+		})
+	});	
+
+  	link_count = menu_links.length + dropdown_links.length;
+
+
+}).run(function(){
+	casper.test.begin('Menu Behaviour', tests, function suite(test) {
+	casper.start(casper.cli.get("target"), function checkTabs() {
+		//Check if all expected menu tabs exist
+		test.assertExists("li.dropdown-suffolk", "Check if Suffolk Cottages Tab Exists.");
+		test.assertExists("li.dropdown-norfolk", "Check if Norfolk Cottages Tab Exists.");
+		test.assertExists("li.dropdown-item.dropdown-coastal", "Check if Coastal Cottages Tab Exists.");
+
+		var other_tabs = casper.evaluate(function(){
+			return Array.prototype.map.call(document.querySelectorAll("ul.main-menu > li:not(.dropdown-item) a"), function(e){
+ 					return e.textContent;
+			});
+		});
+		var expected = ["Special Offers", "What's On", "Holiday Guides", "Sign up for special offers via email"];
+		for(var i = 0; i < expected.length; i++) {
+			var text_content = other_tabs[i];
+			test.assertTrue(text_content == expected[i],  "Check if menu item: " + expected[i] + " exists.");
+		};
+	}).then(function checkDropdowns() {
+		casper.mouse.move("li.dropdown-suffolk");
+		casper.waitUntilVisible("li.dropdown.suffolk", function success() {
```

Comment:
casper.waitUntilVisible("li.dropdown.suffolk", function(){}).then(function(){
...
})
might reduce the nesting, looks fine as long as it's tested, though.